src/python/pants/option/parser.py

```diff
@@ -344,7 +347,8 @@ def add_flag_val(v: Optional[Union[int, float, bool, str]]) -> None:
 
             # If the option is explicitly given, check deprecation and mutual exclusion.
             if val.rank > Rank.HARDCODED:
-                self._check_deprecated(dest, kwargs)
+                print_warning = not parse_args_request.suppress_warnings
```

Comment:
Super nitty: I've recently been using `is True` and `is False` in some places, like 

```python
print_warning = parse_args_request.suppress_warnings is False
```

I find it a bit easier to read. Closer to English.

Reply:
I think this is a good practice actually. It's slightly more verbose than it needs to be, but it is easier to read